pandaserver/dataservice/setupper_atlas_plugin.py

```diff
@@ -0,0 +1,2226 @@
+                    tmp_src_id = job.computingSite
+                elif self.site_mapper.checkCloud(job.getCloud()):
+                    # use cloud's source
+                    tmp_src_id = self.site_mapper.getCloud(job.getCloud())["source"]
+                src_site = self.site_mapper.getSite(tmp_src_id)
+                _, scope_src_site_output = select_scope(src_site, job.prodSourceLabel, job.job_label)
+                src_ddm_id = src_site.ddm_output[scope_src_site_output]
+                # destination
+                tmp_dst_id = job.computingSite
+                tmp_site_spec = self.site_mapper.getSite(job.computingSite)
+                scope_tmp_site_input, _ = select_scope(tmp_site_spec, job.prodSourceLabel, job.job_label)
+                if src_ddm_id != tmp_site_spec.ddm_input[scope_tmp_site_input] and src_ddm_id in tmp_site_spec.setokens_input[scope_tmp_site_input].values():
+                    # direct usage of remote SE. Mainly for prestaging
+                    tmp_dst_id = tmp_src_id
+                    self.logger.debug(f"use remote SiteSpec of {tmp_dst_id} for {job.computingSite}")
+                # use src_ddm_id as dst_ddm_id when it is associated to dest
+                dst_site_spec = self.site_mapper.getSite(tmp_dst_id)
+                scope_dst_input, _ = select_scope(dst_site_spec, job.prodSourceLabel, job.job_label)
+                if dst_site_spec.ddm_endpoints_input[scope_dst_input].isAssociated(src_ddm_id):
+                    dst_ddm_id = src_ddm_id
+                else:
+                    dst_ddm_id = dst_site_spec.ddm_input[scope_dst_input]
+                # check if missing at T1
+                missing_at_t1 = False
+                if job.prodSourceLabel in ["managed", "test"]:
+                    for tmp_lfn in self.disp_file_list[job.dispatchDBlock]["lfns"]:
+                        if job.getCloud() not in self.missing_files_in_t1:
+                            break
+                        if tmp_lfn in self.missing_files_in_t1[job.getCloud()] or tmp_lfn.split(":")[-1] in self.missing_files_in_t1[job.getCloud()]:
+                            missing_at_t1 = True
+                            break
+                    self.logger.debug(f"{job.dispatchDBlock} missing at T1 : {missing_at_t1}")
+                # use DDM
+                if (not self.panda_ddm) and job.prodSourceLabel != "ddm":
+                    # look for replica
+                    ddm_id = src_ddm_id
+                    ddm_id_list = []
+                    # register replica
+                    is_ok = False
+                    if ddm_id != dst_ddm_id or missing_at_t1:
+                        # make list
+                        if job.dispatchDBlock in self.replica_map:
+                            # set DQ2 ID for DISK
+                            if not src_ddm_id.endswith("_DATADISK"):
+                                hot_id = re.sub("_MCDISK", "_HOTDISK", src_ddm_id)
```

Comment:
most of these don't exist anymore

Reply:
needs to be checked in CRIC, but I don't think there is any MCDISK, HOTDISK, MCTAPE